dialer/fastconnect.go

```diff
@@ -76,6 +83,13 @@ func (fcd *fastConnectDialer) DialContext(ctx context.Context, network, addr str
 	return conn, err
 }
 
+func (fcd *fastConnectDialer) Close() {
+	// We don't call Stop on the Dialers themselves here because they are likely
+	// in use by other Dialers, such as the BanditDialer.
+	// Stop all dialing
+	fcd.stopCh <- struct{}{}
```

Comment:
I would make `stopCh` buffered or just close it so this doesn't block until `outerLoop` tries to read from it again.